edgelet/aziot-edged/src/watchdog.rs

```diff
@@ -134,6 +140,34 @@ async fn create_and_start_agent(
     let mut env = agent_env(gen_id, settings, device_info);
     agent_spec.env_mut().append(&mut env);
 
+    let pinfo = if let Some(path) = settings.product_info() {
+        Ok(
+            crate::product_info::ProductInfo::try_load(path).map_err(|err| {
+                EdgedError::from_err(
+                    format!("malformed product info at path {}", path.display()),
+                    err,
+                )
+            })?,
+        )
+    } else {
+        crate::product_info::ProductInfo::from_system()
+    };
+
+    match pinfo {
```

Comment:
Move this (and prev) block to the `agent_env` fn, where the other agent env vars are set.